invenio_search_ui/assets/js/invenio_react_search_ui/App.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * This file is part of Invenio.
+ * Copyright (C) 2020 CERN.
+ *
+ * Invenio is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import React, { Component } from "react";
+import { Container, Grid } from "semantic-ui-react";
+import {
+  ReactSearchKit,
+  SearchBar,
+  BucketAggregation,
+  EmptyResults,
+  Error,
+  ResultsLoader,
+  withState
+} from "react-searchkit";
+import { Results } from "./Results";
+import { InvenioSearchApi } from "react-searchkit";
+
+import "semantic-ui-css/semantic.min.css";
+
+const OnResults = withState(Results);
+
+const resultsPerPageValues = [
+  {
+    text: "10",
+    value: 10
+  },
+  {
+    text: "20",
+    value: 20
+  },
+  {
+    text: "50",
+    value: 50
+  }
+];
+
+export class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { activeIndex: -1 };
+  }
+
+  render() {
+
+    const config = new InvenioSearchApi({
```

Comment:
I would say that this should be configurable too. What about if you have:
```
const searchApi = this.props.searchApi || new InvenioSearchApi(...)
```
this should be possible for all the params that can be passed to the RSK, for at least most of them, see [https://inveniosoftware.github.io/react-searchkit/docs/components/react-searchkit] (I am not sure that the doc there is up-to-date)

Reply:
👍